### ISSUE TITLE
Support announcing provider records

### ISSUE THREAD

**Issue by lidel** [ipfs-shipyard/someguy#11]
Someguy is backed by accelerated DHT client and acts as a proxy for other reframe endpoints.
We want light clients to be able to use `https://routing.delegate.ipfs.io/reframe`  (becked by someguy) for publishing provide records (see https://github.com/protocol/bifrost-infra/issues/2142 for more details about that endpoint).

Reframe's `ProvideRequest` was introduced in https://github.com/ipfs/specs/pull/285

## TODO

- [ ] confirm this is enough to cover both DHT and Reframe puts
  - if not, we need to improve reframe specs in ipfs/specs repo
- [ ] implement provide puts

**Comment by aschmahmann** [comment]
>  confirm this is enough to cover both DHT and Reframe puts

Unfortunately delegated providing to the IPFS Public DHT is impossible due to how the DHT is constructed https://github.com/libp2p/go-libp2p-kad-dht/issues/584. TLDR:
1. The DHT only allows announcing "I have X" not "Alice has X" which means you cannot delegate requests
2. Even if it allowed it there's no way for peer records to announce "Alice has addresses A, B,C" instead addresses are made available by maintaining libp2p connections and utilizing the libp2p `identify` protocol to send addresses over.

So the question is what types of "puts" would you like to do for routing.delegate.ipfs.io? For example, you could decide to only provide to the Indexer network by delegating all the provides to some instance that will hold the records and deal with the semantics there. IIUC @ischasny is working on such a service and implementation at https://github.com/ipfs-shipyard/ipfs-index-provider/pull/1.

A couple thoughts related to integrating with the reframe-based index provider.
1. It could be bundled into someguy, but it turns someguy from a service that requires no persistence or database into one that does ... so I would avoid it and just add one more layer of forwarding/delegation.
2. This service would need to figure out some policy around how it deals with the stored records. Does it store all records forever? Does it require them to be re-put every x hours or else evicted, will the service reach out to the nodes to see if they're alive, hosting some content, etc. If they don't will the service run into problems with the Indexers penalizing it for advertising "bogus" records?

**Comment by gmasgras** [comment]
@lidel is this want still valid/actionable ?

**Comment by lidel** [comment]
@gmasgras no, Reframe was abandoned, so we need to rewrite `someguy` first: https://github.com/ipfs-shipyard/someguy/issues/12, and to have announcement capability, we need to finalize specification of HTTP PUTs. Closing for now.
